src/main/java/org/springframework/hateoas/mvc/ControllerLinkBuilder.java

```diff
@@ -53,17 +50,21 @@
  * @author Kevin Conaway
  * @author Andrew Naydyonock
  * @author Oliver Trosien
+ * @author Josh Ghiloni
+ * @author Greg Turnquist
  */
 public class ControllerLinkBuilder extends LinkBuilderSupport<ControllerLinkBuilder> {
 
 	private static final String REQUEST_ATTRIBUTES_MISSING = "Could not find current request via RequestContextHolder. Is this being called from a Spring MVC handler?";
-	private static final CachingAnnotationMappingDiscoverer DISCOVERER = new CachingAnnotationMappingDiscoverer(

```

Comment:
What's the reason we move the caching of the templates into the `LinkBuilder` and not leave it in the `MappingDiscoverer`. I can see that you only want the raw templates to be cached and not necessarily the values evaluated against the `Environment` (as the latter might change), but I still think this could be solved at construction. Maybe a builder that you hand in an annotation type, that'd wrap the `AMD` with the caching one. If you then add an `Environment` to the builder, it could put the property resolving one in front?